### ISSUE TITLE
Conformance checking?

### ISSUE THREAD

**Issue by TathagataChakraborti** [AI-Planning/p4pp#5]
The P4PP setup reminded me of the [conformance checking problem](https://link.springer.com/content/pdf/10.1007/978-3-319-99414-7_4.pdf) in BPM. 

As an analogy, playing the student's plan forward in the reference model is a type of "token replay", while the trace alignment view has this type of model move versus log move (>>'s) versus synchronous move which is what the feedback of P4PP reminded me of during the demo. Not sure what my point is here, just an observation. 🤷 

<img width="359" alt="Screen Shot 2022-06-23 at 2 44 18 PM" src="https://user-images.githubusercontent.com/4764242/175374299-88a74382-2670-4e27-85d8-b4d13cc39468.png">

Of course, conformance checking also has planning compilations like [this](https://www.diag.uniroma1.it/marrella/papers/SEBD2017.pdf), maybe its possible to write a "Automated Grading as Conformance Checking" paper. 🤔

**Comment by haz** [comment]
Doesn't this align (aha!) closely with the plan rec as planning work? E.g., [this](https://www.ijcai.org/Proceedings/09/Papers/296.pdf). The main difference I see is that it's not a single trace, but rather every trace captured by a domain. It simultaneously looks across them all.

**Comment by TathagataChakraborti** [comment]
Yes indeed! Not that but [this](http://www.cs.toronto.edu/~shirin/Sohrabi-IJCAI-16.pdf), coz we need to drop observations to align.

**Comment by haz** [comment]
I go back to the challenge of this all being flipped. If you want to explore the space of modifications to make two theories align, then you are trying to navigate a solvable space until it is unsolvable. If it was reverse (e.g., what must be change so these theories misalign), then we could package it all up as a big planning problem. Not sure how to promote unplannability as planning :P. Well, there is a way (modeling the successor function in the PDDL itself), but it'd never actually solve.
